### PR TITLE
Rules for irregular verbs and vk

### PR DESCRIPTION
**Pull request by SaltfishAmi**
issue #301  & #303

### CODE REVIEW

**Comment by SaltfishAmi** [comment]
I also found it can't recognize ありえます and ありえません, but I'm just confused after trying ありえる and ありうる since I don't know what 'v2a-s' means...so I wasn't able to make a rule for that.

File: src/deinflect.ts [hunk 1]
```diff
@@ -65,6 +65,9 @@ export const deinflectL10NKeys: { [key: number]: string } = {
 };
 
 const deinflectRuleData: Array<[string, string, number, number]> = [
+  ['いらっしゃいませんでした', 'いらっしゃる', 640, DeinflectReason.PolitePastNegative],
+  ['おっしゃいませんでした', 'おっしゃる', 640, DeinflectReason.PolitePastNegative],
+  ['仰いませんでした', '仰る', 640, DeinflectReason.PolitePastNegative],
```
Comment (birtles) [inline src/deinflect.ts:70]: This array needs to be sorted by the length of the first string.

For entries where the first string has the same length, I've tried to more-or-less keep them in 五十音順.

File: src/deinflect.ts [hunk 2]
```diff
@@ -444,7 +459,11 @@ const deinflectRuleData: Array<[string, string, number, number]> = [
   ['らず', 'る', 640, DeinflectReason.Zu],
   ['らぬ', 'る', 640, DeinflectReason.Negative],
   ['れば', 'る', 7040, DeinflectReason.Ba],
-  ['れる', 'る', 2817, DeinflectReason.Potential],
+  ['れる', 'る', 769, DeinflectReason.Potential],
+  ['来れる', '来る', 2049, DeinflectReason.Potential],
+  ['來れる', '來る', 2049, DeinflectReason.Potential],
+  ['来られる', '来る', 2049, DeinflectReason.Potential],
+  ['來られる', '來る', 2049, DeinflectReason.Potential],
```
Comment (birtles) [inline src/deinflect.ts:466]: I think these two should be `DeinflectReason.PotentialOrPassive`.

File: src/deinflect.ts (hunk 1 above)
Comment (SaltfishAmi) [reply src/deinflect.ts]: I noticed that this array was not completed sorted by the length of the first string, and I resort them as well.

**Comment by birtles** [comment]
Looks great! Thank you!

**Comment by birtles** [comment]
> I also found it can't recognize ありえます and ありえません, but I'm just confused after trying ありえる and ありうる since I don't know what 'v2a-s' means...so I wasn't able to make a rule for that.

I hadn't noticed this.

I think this will be easier to fix once I finish re-doing the data source for the word dictionary since we should have the different verb type annotations in a more structured form.

**Comment by SaltfishAmi** [comment]
> I hadn't noticed this.
> 
> I think this will be easier to fix once I finish re-doing the data source for the word dictionary since we should have the different verb type annotations in a more structured form.

OK, thank you.
By the way, I see there are `en` and `ja` in the `_locales` folder, does this mean I can switch the interface language between English and Japanese?
Because when it comes to grammar terms (like `<-causative`), Japanese is way more clear than English to me.

**Comment by birtles** [comment]
> OK, thank you.
> By the way, I see there are `en` and `ja` in the `_locales` folder, does this mean I can switch the interface language between English and Japanese?
> Because when it comes to grammar terms (like `<-causative`), Japanese is way more clear than English to me.

It should be based on the Firefox language so if you switch Firefox's language then it should use the Japanese terms. Perhaps we should make that configurable, however.

**Comment by SaltfishAmi** [comment]
> It should be based on the Firefox language so if you switch Firefox's language then it should use the Japanese terms. Perhaps we should make that configurable, however.

It will be really nice if it could be configurable.
I'm making a Chinese translation and will create a pr in a few days, maybe after the Dragon Boat Festival.

**Comment by birtles** [comment]
> It will be really nice if it could be configurable.
> I'm making a Chinese translation and will create a pr in a few days, maybe after the Dragon Boat Festival.

Great, thank you so much!

**Pull request merged**